pkg/discovery/util/pod_util.go

```diff
@@ -44,6 +45,14 @@ func IsControllableFromAnnotation(annotations map[string]string) bool {
 	return true
 }
 
+// Returns a boolean that indicates whether the given pod is a daemon pod.  A daemon pod
+// is a pod that resides in a system namespace or was created by a DaemonSet.  These pods
+// are not suspendable, clonable, or movable, and they are not considered when counting
+// customers of a supplier when checking whether the supplier can suspend.
+func Daemon(pod *api.Pod) bool {
+	return isPodCreatedBy(pod, Kind_DaemonSet) || detectors.IsDaemonDetected(pod.Name, pod.Namespace)
```

Comment:
Add some logging here to help future troubleshooting when we detect the daemon.

Reply:
Although the detector functions already generated logging output, I changed those log messages to V4 and added new log message here.